### ISSUE TITLE
FR: ca65: add warnings-as-errors option

### ISSUE THREAD

**Issue by inexorabletash** [cc65/cc65#1563]
I ran into this while moving some code around:
```
ca65 -o target.o target.s
target.s:1234: Warning: "jmp (abs)" across page border
ld65 --config asm.cfg -o target target.o
ld65: Warning: target.s:1234: "jmp (abs)" across page border
```
... but the build didn't fail so I thought _"okay, that's just warning, that's nice"_

And then I read about [what the JMP instruction does when the target straddles page boundaries](http://www.6502.org/tutorials/6502opcodes.html#JMP) and freaked out. This wouldn't cause CI to fail which seems... bad.

I can work around this by failing the build if STDERR is non-empty, but can we get a "treat warnings as errors" option?

I guess this should apply to both ca65 and ld65, and should apply to the current warning level.

**Comment by skeetor** [comment]
IMO this should be an error, not just a warning.

**Comment by mrdudz** [comment]
Adding this option sounds like a good first-time contribution project :)

**Comment by bbbradsmith** [comment]
In lieu of a first-time contributor, I wanted to have this feature too. PR #1999

**Comment by mrdudz** [comment]
MMh is this really how you'd expect this to work? Like this it will first spit out all warnings, and then error out - instead of error out after a warning (like it would do when it would be an error instead), right?

**Comment by bbbradsmith** [comment]
I put the error as late as I could before it generates and output file, that way you get a list of all warnings and not just the first one. I don't think there would be any advantage in stopping only at the first warning?

As in many compilers, we also have in ca65 types of errors that do not immediately terminate the build, so that it can continue and generate a more complete/useful list of errors before eventually terminating with an error. I feel the same applies to warnings-as-errors?

Though, I'm realizing now that OP's request is actually for jmp (abs) across a page to be an error instead of a warning? I'd probably agree with that one. Not sure when it would ever be useful to only warn on that? If it really has to be suppressed, there's always the manual option:
```
.byte $6C ; jmp (abs)
.word pointer
```

**Comment by bbbradsmith** [comment]
I added promotion of jmp (abs) warning to an error to PR #1999 as it seemed convenient to do there instead of a separate PR. This also suppresses the error on 65C02 and later CPUs which don't have that quirk.

**Comment by mrdudz** [comment]
closed via #1999
